### DISCUSSION TITLE
ZepetoScript상속이 잘동작하는건가요?

### DISCUSSION THREAD

**Discussion by Andwhy81kr** [naverz/zepeto-studio-kor#914] [5. Q&A]
제페토스크립트에서 상속이 잘 동작하는건가요?

현재
ItemBase라는 TS를 하나 만들고,
이를 상속받아 
ItemA, ItemB, ItemC.... 라는 TS를 만들었습니다.

몇몇 오브젝트에 제페토스크립트(ItemA,ItemB,ItemC...)를 부착하였고,
이 아이템들을 컨트롤 할 다른 스크립트에서
각 오브젝트에서 GetComponent를 통해 부모클래스(ItemBase)로 컴포넌트를 불러왔습니다.
후에 불러온 컴포넌트의 로그를 찍어보면 null로 컴포넌트를 얻어오지 못하고 있습니다.

단, ItemA, ItemB, ItemC...등으로 부착되어있는 자식 클래스로 GetComponent를 하면 잘동작합니다.

잘못된 접근방법인가요?

**Comment by YSHFM** [comment]
저도 동일한 증상 겪었어서 #858 이 이슈 참고하시면 될 것 같습니다

**Reply by andwhy81** [reply]
감사합니다.
언제 수정될지 걱정이네요
